packages/react/src/components/Tooltip/Tooltip.js

```diff
@@ -263,6 +272,159 @@ class Tooltip extends Component {
     document.addEventListener('keydown', this.handleEscKeyPress, false);
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.direction != this.props.direction) {
+      this.setState({ storedDirection: this.props.direction });
+    }
+    if (prevProps.align != this.props.align) {
+      this.setState({ storedAlign: this.props.align });
+    }
+    if (prevState.open && !this.state.open) {
+      // Reset orientation when closing
+      this.setState({
+        storedDirection: this.props.direction,
+        storedAlign: this.props.align,
+      });
+    }
+  }
+
+  updateOrientation = (params) => {
+    if (this.props.autoOrientation) {
+      const newOrientation = this.getBestDirection(params);
+      const { direction, align } = newOrientation;
+
+      if (direction !== this.state.storedDirection) {
+        console.log(direction);
+        this.setState({ open: false }, () => {
+          this.setState({ open: true, storedDirection: direction });
+        });
+      }
+
+      if (align === 'original') {
+        this.setState({ storedAlign: this.props.align });
+      } else {
+        this.setState({ storedAlign: align });
+      }
+    }
+  };
+
+  getBestDirection = ({
+    menuSize,
+    refPosition = {},
+    offset = {},
+    direction = DIRECTION_BOTTOM,
+    scrollX: pageXOffset = 0,
+    scrollY: pageYOffset = 0,
+    container,
+  }) => {
+    const {
+      left: refLeft = 0,
+      top: refTop = 0,
+      right: refRight = 0,
+      bottom: refBottom = 0,
+    } = refPosition;
+    const scrollX = container.position !== 'static' ? 0 : pageXOffset;
+    const scrollY = container.position !== 'static' ? 0 : pageYOffset;
+    const relativeDiff = {
+      top: container.position !== 'static' ? container.rect.top : 0,
+      left: container.position !== 'static' ? container.rect.left : 0,
+    };
+    const { width, height } = menuSize;
+    const { top = 0, left = 0 } = offset;
+    const refCenterHorizontal = (refLeft + refRight) / 2;
+    const refCenterVertical = (refTop + refBottom) / 2;
+    console.table(container);
```

Comment:
```suggestion
```

Reply:
Oh damn, yes sorry about that, I'll go through and remove any console statements

Reply:
No worries!